### PR TITLE
ripple provider for openlogin adapter

### PR DESCRIPTION
**Pull request by himanshuchawla009**

### CODE REVIEW

File: packages/providers/xrpl-provider/package.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    "prepack": "yarn run build",
+    "pre-commit": "lint-staged --cwd ."
+  },
+  "dependencies": {
+    "@web3auth/base-provider": "^5.0.1",
+    "@web3auth/base": "^5.0.1",
+    "@toruslabs/base-controllers": "^3.0.0",
+   "@toruslabs/openlogin-jrpc": "^4.2.0",
+    "ripple-keypairs": "^1.1.5",
+    "xrpl": "^2.7.0",
```
Comment (chaitanyapotti) [inline packages/providers/xrpl-provider/package.json:27]: make this a peer dependency

File: packages/providers/xrpl-provider/src/providers/privateKeyProviders/interface.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export type XRPLNetwork = "mainnet" | "testnet" | "devnet";
```
Comment (chaitanyapotti) [inline packages/providers/xrpl-provider/src/providers/privateKeyProviders/interface.ts:1]: use const enum

**Review by develoQ — CHANGES_REQUESTED** [review]
In some places and files it should be xrpl, not ripple.

File: packages/providers/xrpl-provider/package.json (hunk 1 above)
Comment (himanshuchawla009) [reply packages/providers/xrpl-provider/package.json:27]: which one and why?

Comment (himanshuchawla009) [reply packages/providers/xrpl-provider/package.json]: fixed

File: packages/providers/xrpl-provider/src/providers/privateKeyProviders/interface.ts (hunk 2 above)
Comment (himanshuchawla009) [reply packages/providers/xrpl-provider/src/providers/privateKeyProviders/interface.ts]: fixed

File: demo/xrpl-react-app /src/services/xrplProvider.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
+import { SafeEventEmitterProvider } from "@web3auth/base";
+import { IWalletProvider } from "./walletProvider";
+import { convertStringToHex, Payment, xrpToDrops } from "xrpl";
+
+const ethProvider = (provider: SafeEventEmitterProvider, uiConsole: (...args: unknown[]) => void): IWalletProvider => {
+  const getAccounts = async () => {
+    try {
+      const accounts = await provider.request<string[]>({
+        method: "ripple_getAccounts"
```
[the remark this answers is not in the document]
Comment (himanshuchawla009) [reply demo/xrpl-react-app /src/services/xrplProvider.ts]: thanks for the review, fixed all you comments.

**Review by develoQ — COMMENTED** [review]
The directory name contains spaces.
`web3auth-web/demo/xrpl-react-app `

should be: `web3auth-web/demo/xrpl-react-app`.

**Pull request merged**